File: tbdfrontend/src/app/dto/Transaction.ts
import {DatePipe, Time} from "@angular/common";

enum TransactionType{
  Credit,
  Debit
}
export interface Transaction {
  transactionId : string,
  customerId : string,
  amount : number,
  roundedAmount : number,
  balance : number,
  roundedBalance : number,
  details : string,
  transactionDate : DatePipe,
  transactionTime : Time,
  transactionType : TransactionType
}

File: src/main/java/org/ece/dto/Transaction.java
package org.ece.dto;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.UUID;

@Data
@Entity
@Table(name = "transaction")
public class Transaction {
    @Id
    private String transactionId = UUID.randomUUID().toString();
    private String customerId;
    private Long amount;
    @Transient
    private double roundedAmount;
    @Transient
    private double roundedBalance;
    private Long balance;
    @Enumerated(EnumType.STRING)
    private TransactionType transactionType;
    private String details;
    private LocalDate transactionDate;
    private LocalTime transactionTime;


}
